indexer/run.sh

```diff
@@ -33,23 +33,96 @@ fetch --nohooks chromium
 
 cd src
 
+mkdir -p out/Build
+export BUILD_DIR=$(readlink -f out/Build)
+
+DATE=$(date -u +%Y%m%d)
+
+echo "target_os = [ 'linux', 'android', 'chromeos', 'fuchsia' ]" >> ../.gclient
+
+gclient sync
+
+gclient runhooks
+
+# $1: the platform name.
+index() {
+  echo "Indexing for $1"
+
+  ninja -C $BUILD_DIR -t targets all | grep -i '^gen/' | grep -E "\.(cpp|h|inc|cc)\:" | cut -d':' -f1 | xargs autoninja -C $BUILD_DIR
+
+  tools/clang/scripts/generate_compdb.py -p $BUILD_DIR > compile_commands.json
+
+  $CLANGD_INDEXER --executor=all-TUs compile_commands.json > /chrome-$1.idx
+
+  7z a chrome-index-$1-$DATE.zip /chrome-$1.idx
+
+  # Clean up the build directory afterwards.
+  rm -rf $BUILD_DIR
+}
+
+# --- Linux ---
+
+PLATFORM="linux"
+
 # Remove snapcraft from dependency list: installing it is not feasible inside
 # Docker.
 sed -i '/if package_exists snapcraft/,/fi/d' ./build/install-build-deps.sh
 ./build/install-build-deps.sh
 
-gclient runhooks
+gn gen --args='target_os="linux"' $BUILD_DIR
+
+index $PLATFORM
+
+# --- Linux Chromecast ---
+
+PLATFORM="linux-chromecast"
+
+gn gen --args='target_os="linux" is_chromecast=true' $BUILD_DIR
+
+index $PLATFORM
+
+# --- Android ---
+
+PLATFORM="android"
+
+build/install-build-deps-android.sh
+
+gn gen --args='target_os="android"' $BUILD_DIR
+
+index $PLATFORM
+
+# --- Android Chromecast ---
+
+PLATFORM="android-chromecast"
+
+gn gen --args='target_os="android" is_chromecast=true' $BUILD_DIR
+
+index $PLATFORM
+
+# --- Fuchsia ---
+
+PLATFORM="fuchsia"
+
+gn gen --args='target_os="fuchsia"'
+
+index $PLATFORM
 
-gn gen out/Default
+# --- ChromeOS ---
 
-ninja -C out/Default/ -t targets all | grep -i '^gen/' | grep -E "\.(cpp|h|inc|cc)\:" | cut -d':' -f1 | xargs autoninja -C out/Default
+PLATFORM="chromeos"
 
-tools/clang/scripts/generate_compdb.py -p out/Default > compile_commands.json
+gn gen --args='target_os="chromeos"' $BUILD_DIR
 
-$CLANGD_INDEXER --executor=all-TUs compile_commands.json > /chrome.idx
+index $PLATFORM
 
-7z a chrome-index-$(date -u +%Y%m%d).zip /chrome.idx
+# -- Finish the job ---
 
 CURRENT_COMMIT=$(git rev-parse --short HEAD)
 
-gh release create --repo clangd/chrome-remote-index --title="Index at $(date -u +%Y-%m-%d)" --notes="Index with Default config options at $CURRENT_COMMIT commit." $CURRENT_COMMIT chrome-index-$(date -u +%Y%m%d).zip
+gh release create --repo clangd/chrome-remote-index --title="Index at $DATE" --notes="Index with at $CURRENT_COMMIT commit." $CURRENT_COMMIT \
+  chrome-index-linux-$DATE.zip \
```

Comment:
what about uploading artifacts as part of the `index` call ? that way we can start serving a fresher index for more active build configs, rather than introducing N hours delay for all.